_inc/lib/debugger/class-jetpack-cxn-test-base.php

```diff
@@ -49,21 +51,91 @@ public function __construct() {
 	/**
 	 * Adds a new test to the Jetpack Connection Testing suite.
 	 *
+	 * @since 7.1.0
+	 * @since 7.3.0 Adds name parameter and returns WP_Error on failure.
+	 *
 	 * @param callable $callable Test to add to queue.
-	 * @param array    $groups Testing groups to add test to.
+	 * @param string   $name Unique name for the test.
+	 * @param string   $type   Optional. Core Site Health type: 'direct' if test can be run during initial load or 'async' if test should run async.
+	 * @param array    $groups Optional. Testing groups to add test to.
 	 *
-	 * @return bool True if successfully added. False for a failure.
+	 * @return mixed True if successfully added. WP_Error on failure.
 	 */
-	public function add_test( $callable, $groups = array( 'default' ) ) {
+	public function add_test( $callable, $name, $type = 'direct', $groups = array( 'default' ) ) {
+		if ( is_array( $name ) ) {
+			// Pre-7.3.0 method passed the $groups parameter here.
+			return new WP_Error( __( 'add_test arguments changed in 7.3.0. Please reference inline documentation.', 'jetpack' ) );
+		}
+		if ( array_key_exists( $name, $this->tests ) ) {
+			return new WP_Error( __( 'Test names must be unique.', 'jetpack' ) );
+		}
 		if ( is_callable( $callable ) ) {
-			$this->tests[] = array(
+			$this->tests[ $name ] = array(
 				'test'  => $callable,
 				'group' => $groups,
+				'type'  => $type,
 			);
 			return true;
 		}
 
-		return false;
+		return new WP_Error( __( 'Tests must be valid PHP callables.', 'jetpack' ) );
```

Comment:
not a big deal, but it might be a bit more readable if all the error checks will be in the begining of method: 

```suggestion
		if ( ! is_callable( $callable ) ) {
		        return new WP_Error( __( 'Tests must be valid PHP callables.', 'jetpack' ) );
		}
		
		$this->tests[ $name ] = array(
			'test'  => $callable,
			'group' => $groups,
			'type'  => $type,
		);
		return true;
```

Reply:
Added via 1dc1e79ab